python/pyarrow/tests/test_table.py

```diff
@@ -2301,6 +2301,10 @@ def test_table_from_pydict(cls):
     with pytest.raises(TypeError):
         cls.from_pydict({'a': [1, 2, 3]}, schema={})
 
+    # With dicts containing NaNs
```

Comment:
This test isn't related to `from_pydict`, we should either make it a purely C++ test as @kou suggested, or add it to an equality related test case.